cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -156,30 +185,84 @@ func gitIndex(o *indexArgs, runCmd func(*exec.Cmd) error) error {
 		return err
 	}
 
-	fetchStart := time.Now()
+	var fetchDuration time.Duration
+	successfullyFetchedCommitsCount := 0
+	allFetchesSucceeded := true
 
-	// We shallow fetch each commit specified in zoekt.Branches. This requires
-	// the server to have configured both uploadpack.allowAnySHA1InWant and
-	// uploadpack.allowFilter. (See gitservice.go in the Sourcegraph repository)
-	fetchArgs := []string{"-C", gitDir, "-c", "protocol.version=2", "fetch", "--depth=1", o.CloneURL}
-	var commits []string
-	for _, b := range o.Branches {
-		commits = append(commits, b.Version)
+	defer func() {
+		success := strconv.FormatBool(allFetchesSucceeded)
+		name := repoNameForMetric(o.Name)
+		metricFetchDuration.WithLabelValues(success, name).Observe(fetchDuration.Seconds())
+	}()
+
+	var fetchCommits = func(commits []string) error {
+		// We shallow fetch each commit specified in zoekt.Branches. This requires
+		// the server to have configured both uploadpack.allowAnySHA1InWant and
+		// uploadpack.allowFilter. (See gitservice.go in the Sourcegraph repository)
+		fetchArgs := []string{"-C", gitDir, "-c", "protocol.version=2", "fetch", "--depth=1", o.CloneURL}
+		fetchArgs = append(fetchArgs, commits...)
+
+		cmd = exec.CommandContext(ctx, "git", fetchArgs...)
+		cmd.Stdin = &bytes.Buffer{}
+
+		start := time.Now()
+		err := runCmd(cmd)
+		fetchDuration += time.Since(start)
+
+		if err != nil {
+			allFetchesSucceeded = false
+			return err
+		}
+
+		successfullyFetchedCommitsCount += len(commits)
+		return nil
 	}
-	fetchArgs = append(fetchArgs, commits...)
 
-	cmd = exec.CommandContext(ctx, "git", fetchArgs...)
-	cmd.Stdin = &bytes.Buffer{}
+	var branchCommits []string
+	for _, b := range o.Branches {
+		branchCommits = append(branchCommits, b.Version)
+	}
 
-	err = runCmd(cmd)
-	fetchDuration := time.Since(fetchStart)
+	err = fetchCommits(branchCommits)
 	if err != nil {
-		metricFetchDuration.WithLabelValues("false", repoNameForMetric(o.Name)).Observe(fetchDuration.Seconds())
 		return err
 	}
 
-	metricFetchDuration.WithLabelValues("true", repoNameForMetric(o.Name)).Observe(fetchDuration.Seconds())
-	debug.Printf("fetched git data for %q (%d commit(s)) in %s", o.Name, len(commits), fetchDuration)
+	if o.UseDelta {
+		// Try fetching prior commits for delta builds
+		// If we're unable to fetch prior commits, we continue anyway
+		// knowing that zoekt-git-index will fall back to a "full" normal build
+
+		var priorCommits []string
+
+		existingRepository, found, err := findRepositoryMetadata(o)
+		if err != nil {
+			return fmt.Errorf("delta build: failed to get repository metadata: %w", err)
+		}
+
+		if found {
+			for _, b := range existingRepository.Branches {
+				priorCommits = append(priorCommits, b.Version)
+			}
+		}
+
+		if len(priorCommits) > 0 {
```

Comment:
will you ever have len(priorCommits) == 0 if found? Can you just combine the two conditionals here?